kong/api/endpoints.lua

```diff
@@ -303,12 +303,17 @@ end
 local function get_collection_endpoint(schema, foreign_schema, foreign_field_name, method)
   return not foreign_schema and function(self, db, helpers)
     local next_page_tags = ""
+    local next_page_size = ""
 
     local args = self.args.uri
     if args.tags then
       next_page_tags = "&tags=" .. escape_uri(type(args.tags) == "table" and args.tags[1] or args.tags)
     end
 
+    if args.size then
+      next_page_size = "&size=" .. escape_uri(type(args.size) == "number" and args.size[1] or args.size)
```

Comment:
It is a logic error, we can not access `[1]` of `number` type.

Reply:
@chronolaw Sorry, my mistake! I validated now only if the type is numeric, I think it should solve